radixdlt/src/main/java/com/radixdlt/consensus/AddressBookValidatorSetProvider.java

```diff
@@ -59,15 +69,35 @@ public AddressBookValidatorSetProvider(ECPublicKey selfKey, AddressBook addressB
 			).distinct().collect(Collectors.toList()))
 			.filter(peers -> peers.size() == fixedNodeCount)
 			.firstOrError()
-			.map(peers -> ValidatorSet.from(
+			.map(peers ->
 				peers.stream()
+					.sorted(Comparator.comparing(ECPublicKey::euid))
 					.map(p -> Validator.from(p, UInt256.ONE))
-					.collect(Collectors.toList())
-			))
+					.collect(ImmutableList.toImmutableList())
+			)
 			.cache();
 	}
 
 	public ValidatorSet getValidatorSet(long epoch) {
-		return validatorSet.blockingGet();
+		ImmutableList<Validator> validators = validatorList.blockingGet();
+
+		Builder<Validator> validatorSetBuilder = ImmutableList.builder();
+		Random random = new Random(epoch);
+		List<Integer> indices = IntStream.range(0, validators.size()).boxed().collect(Collectors.toList());
+		for (long i = 0; i < epoch; i++) {
+			random.nextInt(validators.size());
+		}
+		int randInt = random.nextInt(validators.size());
+		int validatorSetSize = randInt + 1;
```

Comment:
Why do we want random validator sets of varying sizes? I thought we just wanted the constituent validators to be random, but the set be of a fixed size? Did I misunderstand or has that changed?

Reply:
this isn't the final configuration. Just the configuration for the time being for testing.

Reply:
It's just random for now not as the final configuration